src/commands/authorize/authorizeCommand.ts

```diff
@@ -0,0 +1,71 @@
+import { AuthorizeInterface } from "../../interfaces/authorize";
+
+export default class AuthorizeCommand {
```

Comment:
I would strongly advise against using this pattern for creating an `AuthorizeCommand`. The class doesn't add any value compared to using an object that satisfies an interface. We need to add a lot of code (which will have to be maintained) to be able to create the object, read its state, and (de)serialize it). In other languages this boilerplate would actually be automatically generated by a preprocessor to avoid mistakes.
This pattern also introduces multiple issues:
1. if I want to serialize this class I'd have to JSON.stringify(authorizeCommand.toJSON()) instead of just JSON.stringify(authorizeCommand)
2. I can't do a `new AuthorizeCommand` without a cookie (it's optional) but with a sessionToken. Unless I pass undefined to cookie but that's not nice. We generally recommend a max number of params in a function of 2 to 3 and then pass an object for optional parameters. Now we have a lot more boilerplate code to write.
3. How would I extend this object without modifying code? If we're using a plain object I could extend my interface, but here I need to change the constructor, change the .toJSON function, add a getter etc.

Reply:
If we're looking for immutability (I see that all class fields are private) we can simply use `Object.freeze()` on the command. And if we're looking to validate the shape of the object, the interface will make sure that our object is valid at build time. Also, if we need to validate the object at runtime, then I'd simply suggest adding a validator function using `yup` for instance